### ISSUE TITLE
Make documentation rustier/ NumPy compliant

### ISSUE THREAD

**Issue by MicahGale** [idaholab/MontePy#559]
In [Rust](https://doc.rust-lang.org/rustdoc/how-to-write-documentation.html#documenting-components) there is an expectation that doc strings will include examples.

This could be a good idea to start including more and more in our doc strings.

Along with this we would need to implement [doctest](https://docs.pytest.org/en/7.1.x/how-to/doctest.html) in pytest. 

Is this a good paradigm?

**Comment by tjlaboss** [comment]
doctest is now implemented, opening this up

**Comment by MicahGale** [comment]
Right. Probably need to create a new convention for docstrings including examples as well a reference to the MCNP manuals.

**Comment by MicahGale** [comment]
Should I add this to the big todo list for the 1.0.0 alpha release?

**Comment by MicahGale** [comment]
As mentioned in #685, we will be complying with `numpydoc` standards. See: https://numpydoc.readthedocs.io/en/latest/format.html.

**Comment by tjlaboss** [comment]
Related scope: #413
